views/components/ConfigurationsList/index.jsx

```diff
@@ -415,268 +264,69 @@ const ConfigurationsList = ({ user, classes, theme }) => {
-                      tooltip="View"
-                    >
-                      <FullView />
-                    </IconButton>
-                  )}
-                  {ownsConfig ? (
-                    <IconButton
-                      iconStyle={{ color: 'rgba(0, 0, 0, 0.54)' }}
-                      tooltipPosition="top-center"
-                      tooltip="Share"
-                      onClick={() =>
-                        openSearchUsers(
-                          item,
-                          configs[item]['_id'],
-                          configs[item]['readers'],
-                          configs[item]['owner']
-                        )
-                      }
-                    >
-                      <Share />
-                    </IconButton>
-                  ) : (
-                    <IconButton
-                      iconStyle={{ color: 'rgba(0, 0, 0, 0.54)' }}
-                      tooltipPosition="top-center"
-                      tooltip="Duplicate"
-                      onClick={() => copyConfig(configs[item])}
-                    >
-                      <Copy />
-                    </IconButton>
-                  )}
-                </div>
-                <FormControlLabel
-                  control={
-                    <Switch
-                      style={{
-                        width: 'auto',
-                      }}
-                      labelStyle={{ color: 'rgba(0, 0, 0, 0.54)' }}
-                      checked={
-                        'config' in preference
-                          ? configs[item]['_id'] == preference['config']
-                          : false
-                      }
-                      onChange={(e, isInputChecked) =>
-                        changeDefaultConfig(e, isInputChecked, item)
-                      }
-                    />
-                  }
-                  label="Default"
-                />
-              </div>
-            </CardActions>
-          </Card>
-        )
-      }
-    }
-    return cards
-  }
-  const shareWithUsers = () => {
-    const { _id } = state.selectedConfig
+  const shareWithUsers = async () => {
+    const { _id } = sharedWithState.selectedConfig
     const configAttributes = {
-      readers: state.shared.map((sharedWith) => sharedWith.value),
+      readers: sharedWithState.shared.map((sharedWith) => sharedWith.value),
     }
-
-    return window
-      .fetch(apiRoutes.configurations.userConfiguration(user.uid, _id), {
-        method: 'PUT',
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        credentials: 'same-origin',
-        body: JSON.stringify(configAttributes),
-      })
-      .then((response) => {
-        if (response.status === 200) {
-          setState((prevState) => {
+    const response = await UserConfigurationsModel.update(
+      uid,
+      _id,
+      configAttributes
+    )
+    if (response.status === 200) {
+      setConfigurations(
```

Comment:
we can use `loadAllConfigurations` here which will take care of loading the new configurations from the API and storing them in state.

Reply:
[d31c716](https://github.com/AMP-SCZ/dpdash/pull/310/commits/d31c716b584e340b90068954fc940c49761e52fe)